src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -436,12 +436,7 @@ private function addValidationSection(ArrayNodeDefinition $rootNode)
                     ->info('validation configuration')
                     ->canBeEnabled()
                     ->children()
-                        ->scalarNode('cache')
-                            ->beforeNormalization()
-                                // Can be removed in 3.0, once ApcCache support is dropped
-                                ->ifString()->then(function ($v) { return 'apc' === $v ? 'validator.mapping.cache.apc' : $v; })
```

Comment:
Removing the feature because it's not documented, mays be a point.
BUT, If someone use the configuration `cache: apc` with symfony 2.7, he never been warned about the deprecation. Upgrading to Symfony 3 lead to an error due to the missing class.
He may expect we fix the bug, instead of removing the feature.

Just my 2 cents and playing the devil's advocate. I understand the point, to cleanup this code :wink: